skyportal/models.py

```diff
@@ -489,6 +489,28 @@ def desi_dr8_url(self):
             f"&dec={self.dec}&size=200&layer=dr8&pixscale=0.262&bands=grz"
         )
 
+    @property
+    def panstarrs_url(self):
+        """Construct URL for public PanSTARRS-1 (PS1) cutout.
+
+        The cutout service doesn't allow directly querying for an image; the
+        best we can do is request a page that contains a link to the image we
+        want (in this case a combination of the green/blue/red filters).
+        """
+        try:
+            ps_query_url = (
+                f"http://ps1images.stsci.edu/cgi-bin/ps1cutouts"
+                f"?pos={self.ra}+{self.dec}&filter=color&filter=g"
+                f"&filter=r&filter=i&filetypes=stack&size=250"
+            )
+            response = requests.get(ps_query_url)
+            match = re.search(
+                'src="//ps1images.stsci.edu.*?"', response.content.decode()
```

Comment:
I'd probably have opted for BeautifulSoup4, to also avoid the replace below which feels slightly brittle, but I'll leave it up to you.

Reply:
I just resurrected whatever Brett wrote back in the day -- this is his implementation